k4e-agent-sti-upgrade.sh

```diff
@@ -0,0 +1,194 @@
+    usage
+    exit 1
+fi
+
+if [[ ! -z $VERBOSE ]]; then
+    echo "Building ${IMAGE_NAME} image for Operator's HTTP-API ${HTTP_API}"
+    set -xv
+fi
+
+# export for use in templates
+export IMAGE_NAME
+export HTTP_API
+export RAND_SUFFIX=$((1 + $RANDOM % 10))
+PACKAGES_REPO=k4e-repo-$RAND_SUFFIX
+PACKAGE_REPO_URL=http://$IMAGE_SERVER_ADDRESS/$PACKAGES_REPO
+BLUEPRINT_NAME=$IMAGE_NAME
+IMAGE_UPGRADE_FOLDER=/var/www/html/$IMAGE_NAME-upgrade
+IMAGE_UPGRADE_URL=http://$IMAGE_SERVER_ADDRESS/$IMAGE_NAME-upgrade
+
+#---------------------------------
+# Cleanup before running
+#---------------------------------
+rm -rf ~/rpmbuild/*
+rm -rf /var/www/html/k4e-repo*
+rm -rf /home/builder/yggdrasil
+rm -rf /home/builder/k4e-device-worker
+rm -rf /home/builder/prometheus-node_exporter-rpm
+rm -rf /var/www/html/$IMAGE_UPGRADE_FOLDER
+rm -rf /var/www/html/$IMAGE_NAME-upgrade
+# need to remove all sources
+composer-cli sources delete agent
+
+#-----------------------------------
+# Build packages for k4e from source (change here the branches)
+#-----------------------------------
+# Build yggdrasil rpm
+git clone https://github.com/jakub-dzon/yggdrasil.git /home/builder/yggdrasil
+cd /home/builder/yggdrasil
+export CGO_ENABLED=0
+
+export ARCH=$(uname -i)
+GOPROXY=proxy.golang.org,direct PWD=$PWD spec=$PWD outdir=$PWD make -f .copr/Makefile srpm
+rpm -ihv $(ls -ltr yggdrasil-*.src.rpm | tail -n 1 | awk '{print $NF}')
+if [ "$ARCH" = "aarch64" ]; then
+    # Turn ELF binary stripping off in %post
+    sed -i '1s/^/%global __os_install_post %{nil}/' ~/rpmbuild/SPECS/yggdrasil.spec
+fi
+rpmbuild -bb ~/rpmbuild/SPECS/yggdrasil.spec --target $ARCH
+
+# Build k4e-device-worker rpm
+git clone https://github.com/bardielle/k4e-device-worker.git /home/builder/k4e-device-worker
+cd /home/builder/k4e-device-worker
+git checkout oslifecycle
+
+if [ "$ARCH" = "aarch64" ]; then
+    make build-arm64
+    make rpm-arm64
+else
+    make build
+    make rpm
+fi
+
+# Build Prometheus node_exporter rpm
+git clone https://github.com/jakub-dzon/prometheus-node_exporter-rpm.git /home/builder/prometheus-node_exporter-rpm
+cd /home/builder/prometheus-node_exporter-rpm
+make rpm
+
+#---------------------------
+# Create packages repository
+#---------------------------
+mkdir /var/www/html/$PACKAGES_REPO
+cp /root/rpmbuild/RPMS/${ARCH}/*.${ARCH}.rpm /var/www/html/$PACKAGES_REPO/
+createrepo /var/www/html/$PACKAGES_REPO/
+
+#-----------------------------------
+# Create and publish rhel4edge new commit
+#-----------------------------------
+cd /home/builder/r4e
+
+export REPO_URL=$IMAGE_BASE_URL/repo
+
+# make sure image does not already exist
+if [ -e "$IMAGE_UPGRADE_FOLDER" ] ; then
+    echo "$IMAGE_UPGRADE_FOLDER already exists"
+    exit 1
+fi
+
+#-----------------------------------
+# create source resource if rpm repository URL is provided
+#-----------------------------------
+cat << EOF > repo.toml
+id = "agent-$RAND_SUFFIX"
```

Comment:
let's add k4e-prefix in this context so it will be identifiable from as k4e resource.

also, the `agent-$RAND_SUFFIX` repeats 3 times, let's set it in a variable and reuse it.